wrappers/awskms/awskms.go

```diff
@@ -244,10 +263,9 @@ func (k *Wrapper) Decrypt(_ context.Context, in *wrapping.BlobInfo, opt ...wrapp
 		plaintext = output.Plaintext
 
 	case AwsKmsEnvelopeAesGcmEncrypt:
-		// KeyId is not passed to this call because AWS handles this
-		// internally based on the metadata stored with the encrypted data
 		input := &kms.DecryptInput{
 			CiphertextBlob: in.KeyInfo.WrappedKey,
+			KeyId: keyID,
```

Comment:
I'm not sure this is the correct key to pass to the decrypt operation. This could be a key alias or a fully qualified ARN. In the case of key alias, the key could have been rotated underneath that no longer matches the key used to encrypt the value. I think you'll want to use the in.KeyInfo.KeyID instead. I believe this will only work for wrapped keys though.